simpleclient_httpserver/src/main/java/io/prometheus/client/exporter/HTTPServer.java

```diff
@@ -36,6 +36,17 @@
  * </pre>
  * */
 public class HTTPServer {
+
+    static {
+        if (!System.getProperties().containsKey("sun.net.httpserver.maxReqTime")) {
+            System.setProperty("sun.net.httpserver.maxReqTime", "60");
+        }
+
+        if (!System.getProperties().containsKey("sun.net.httpserver.maxRspTime")) {
+            System.setProperty("sun.net.httpserver.maxRspTime", "600");
+        }
+    }
+
```

Comment:
I tried adding a test for this as well, like this:

```diff
diff --git simpleclient_httpserver/src/test/java/io/prometheus/client/exporter/TestHTTPServer.java simpleclient_httpserver/src/test/java/io/prometheus/client/exporter/TestHTTPServer.java
index c9f6bee..f02dbd4 100644
--- simpleclient_httpserver/src/test/java/io/prometheus/client/exporter/TestHTTPServer.java
+++ simpleclient_httpserver/src/test/java/io/prometheus/client/exporter/TestHTTPServer.java
@@ -13,6 +13,7 @@ import java.util.zip.GZIPInputStream;
 import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
+import sun.net.httpserver.ServerImplHelper;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.fail;
@@ -147,4 +148,14 @@ public class TestHTTPServer {
     String response = requestWithCompression("/-/healthy", "");
     assertThat(response).contains("Exporter is Healthy");
   }
+
+  @Test
+  public void testRequestTimeout() {
+    assertThat(ServerImplHelper.getRequestTimeout()).isEqualTo(60000);
+  }
+
+  @Test
+  public void testResponseTimeout() {
+    assertThat(ServerImplHelper.getResponseTimeout()).isEqualTo(600000);
+  }
 }
diff --git simpleclient_httpserver/src/test/java/sun/net/httpserver/ServerImplHelper.java simpleclient_httpserver/src/test/java/sun/net/httpserver/ServerImplHelper.java
new file mode 100644
index 0000000..f841435
--- /dev/null
+++ simpleclient_httpserver/src/test/java/sun/net/httpserver/ServerImplHelper.java
@@ -0,0 +1,17 @@
+package sun.net.httpserver;
+
+/**
+ * Helper class to be able to read package-internal fields from {@link ServerImpl}.
+ *
+ * @author Per Lundberg
+ */
+public class ServerImplHelper {
+
+    public static long getRequestTimeout() {
+        return ServerImpl.MAX_REQ_TIME;
+    }
+
+    public static long getResponseTimeout() {
+        return ServerImpl.MAX_RSP_TIME;
+    }
+}
```

However, this fails with an exception: `java.lang.IllegalAccessError: tried to access class sun.net.httpserver.ServerImpl from class sun.net.httpserver.ServerImplHelper`. The `sun.net.httpserver` classes are really not written with (external, non-JDK) testability in mind... :unamused: